### PR TITLE
Write commit build history cache with indented json

### PR DESCRIPTION
**Pull request by freewheel70**
<https://dev.azure.com/ceapex/Engineering/_workitems/edit/572576/> 

Format build-history.json with indent for better diff debug experience.

### CODE REVIEW

File: src/docfx/lib/process/ProcessUtility.cs [hunk 1]
```diff
@@ -123,12 +131,12 @@ public static void ReadFile(string path, Action<Stream> read)
...
     /// When used together with <see cref="ReadJsonFile{T}(string)"/>, provides inter-process synchronized access to the file.
     /// </summary>
-    public static void WriteJsonFile<T>(string path, T data)
+    public static void WriteJsonFile<T>(string path, T data, bool withIndent = false)
     {
         using (InterProcessMutex.Create(path))
         using (var fs = new FileStream(path, FileMode.Create, FileAccess.Write, FileShare.None, 1024, FileOptions.SequentialScan))
         {
-            JsonSerializer.Serialize(fs, data, s_jsonOptions);
+            JsonSerializer.Serialize(fs, data, withIndent ? s_indentedJsonOptions : s_jsonOptions);
```
Comment (herohua) [inline src/docfx/lib/process/ProcessUtility.cs:139]: Can we simply leverage https://github.com/dotnet/docfx/blob/d9d4b7f28d8f0ac0969b3ca59fd63378090821e0/src/docfx/lib/json/JsonUtility.cs#L100-L100?

Comment (freewheel70) [reply src/docfx/lib/process/ProcessUtility.cs]: @herohua Good point, I just refactor the code.

**Review by herohua — APPROVED** [review]

**Review by yufeih — COMMENTED** [review]
Is this file too big to format in vscode?

**Comment by freewheel70** [comment]
> Is this file too big to format in vscode?

@yufeih Yes, for `azure-docs-pr` , the build-history.json file is 78.3MB for now which is too big to format in VSCode.
And after formatting with indent the size increases  to 103MB ( about 130% ), but we could directly see the diff in web UI.

**Comment by yufeih** [comment]
> > Is this file too big to format in vscode?
> 
> @yufeih Yes, for `azure-docs-pr` , the build-history.json file is 78.3MB for now which is too big to format in VSCode. And after formatting with indent the size increases to 103MB ( about 130% ), but we could directly see the diff in web UI.

How big are the other files written using the same method, if they are all below the critical you think reasonable, we probably don't need a flag and just write them all as intended.

**Comment by freewheel70** [comment]
>  How big are the other files written using the same method,

@yufeih Thanks for reminding, I just checked the code, the other 2 files are using the same method (i.e. msgraph-users.json and github-users.json) and their sizes are quite small (i.e. 116KB and 1347KB). So I updated the code to write them all as intended

**Review by yufeih — APPROVED** [review]

**Comment by herohua** [comment]
<details><summary>🚗🌳💥🤕🚑<a href='https://github.com/MicrosoftDocs/powerbi-docs-pr.de-DE'>powerbi-docs-pr.de-DE</a>(00:00:00, 0MB)</summary>

```
System.InvalidOperationException: '"D:\a\_work\1\RegressionTest\docfx.exe" restore --log "D:/docfx.TestData\regression-test.powerbi-docs-pr.de-DE\.temp\.errors.log" --verbose --stdin' failed in directory 'D:/docfx.TestData\regression-test.powerbi-docs-pr.de-DE\powerbi-docs-pr.de-DE' with exit code 1, message: 
 
   at Microsoft.Docs.Build.RegressionTest.Exec(String fileName, String arguments, String stdin, String cwd, Dictionary`2 env, Boolean ignoreError, Boolean redirectStandardError, Int32[] allowExitCodes, String[] secrets) in D:\a\_work\1\s\test\docfx.RegressionTest\RegressionTest.cs:line 411
   at Microsoft.Docs.Build.RegressionTest.RestoreDependency(String repositoryPath, String docfxConfig, String outputPath) in D:\a\_work\1\s\test\docfx.RegressionTest\RegressionTest.cs:line 239
   at Microsoft.Docs.Build.RegressionTest.Build(RegressionTestResult testResult, String repositoryPath, String outputPath, Options opts, String docfxConfig) in D:\a\_work\1\s\test\docfx.RegressionTest\RegressionTest.cs:line 254
   at Microsoft.Docs.Build.RegressionTest.Test(Options opts, String workingFolder) in D:\a\_work\1\s\test\docfx.RegressionTest\RegressionTest.cs:line 173
   at Microsoft.Docs.Build.RegressionTest.Run(Options opts) in D:\a\_work\1\s\test\docfx.RegressionTest\RegressionTest.cs:line 64

```

</details>

**Comment by herohua** [comment]
<details><summary>🧭<a href='https://github.com/MicrosoftDocs/PowerShell-Docs'>PowerShell-Docs</a>(00:01:21.5544999 | exceed 75s, 4137MB)</summary>



</details>

**Comment by herohua** [comment]
<details><summary>⚠<a href='https://github.com/MicrosoftDocs/learn-pr'>learn-pr</a>(00:01:32.3996094, 8159 more diff, 8104MB)</summary>

```diff
diff --git "a/D:/docfx.TestData\\regression-test.learn-pr\\output/learn-pr/.publish.json" "b/D:/docfx.TestData\\regression-test.learn-pr\\.temp/learn-pr/.publish.json"
index 6187b30..92aadf9 100644
--- "a/D:/docfx.TestData\\regression-test.learn-pr\\output/learn-pr/.publish.json"
+++ "b/D:/docfx.TestData\\regression-test.learn-pr\\.temp/learn-pr/.publish.json"
@@ -146496,6 +146496,9 @@
       "canonical_url": "https://docs.microsoft.com/en-us/learn/modules/5g-business-models/1-introduction",
       "document_id": "94145aa7-8117-87c5-d1b8-0d0e08fea2bc",
       "document_version_independent_id": "d18fef74-ec27-eff4-909a-0a5c13e8c238",
+      "github_contributors": [
+        "aysemutlu"
+      ],
       "updated_at": "2021-11-08 12:20 AM",
       "content_git_url": "https://github.com/MicrosoftDocs/learn/blob/main/learn-pr/student-evangelism/5g-business-models/1-introduction.yml",
       "original_content_git_url": "https://github.com/MicrosoftDocs/learn-pr/blob/live/learn-pr/student-evangelism/5g-business-models/1-introduction.yml",
@@ -146554,6 +146557,9 @@
       "canonical_url": "https://docs.microsoft.com/en-us/learn/modules/5g-business-models/2-business-functions",
       "document_id": "830e9803-62c9-a3d7-a7e9-b3d847b1d44a",
       "document_version_independent_id": "d5f0fd26-01b5-d0e0-14fd-223559447b7c",
+      "github_contributors": [
+        "aysemutlu"
+      ],
       "updated_at": "2021-11-08 12:20 AM",
       "content_git_url": "https://github.com/MicrosoftDocs/learn/blob/main/learn-pr/student-evangelism/5g-business-models/2-business-functions.yml",
       "original_content_git_url": "https://github.com/MicrosoftDocs/learn-pr/blob/live/learn-pr/student-evangelism/5g-business-models/2-business-functions.yml",
@@ -146612,6 +146618,9 @@
       "canonical_url": "https://docs.microsoft.com/en-us/learn/modules/5g-business-models/3-how-5g-works",
       "document_id": "6921f8cc-a699-b663-5023-31c888fe5ba9",
       "document_version_independent_id": "079f532f-6cc5-9e83-9af6-cfc40dd0cb1f",
+      "github_contributors": [
+        "aysemutlu"
+      ],
       "updated_at": "2021-11-08 12:20 AM",
       "content_git_url": "https://github.com/MicrosoftDocs/learn/blob/main/learn-pr/student-evangelism/5g-business-models/3-how-5g-works.yml",
       "original_content_git_url": "https://github.com/MicrosoftDocs/learn-pr/blob/live/learn-pr/student-evangelism/5g-business-models/3-how-5g-works.yml",
@@ -146670,6 +146679,9 @@
       "canonical_url": "https://docs.microsoft.com/en-us/learn/modules/5g-business-models/4-when-use",
       "document_id": "d4417019-9c4d-1385-150e-373543560d11",
       "document_version_independent_id": "6fc6adcd-4f20-287c-c33e-e0687c1d2b90",
+      "github_contributors": [
+        "aysemutlu"
+      ],
       "updated_at": "2021-11-08 12:20 AM",
       "content_git_url": "https://github.com/MicrosoftDocs/learn/blob/main/learn-pr/student-evangelism/5g-business-models/4-when-use.yml",
       "original_content_git_url": "https://github.com/MicrosoftDocs/learn-pr/blob/live/learn-pr/student-evangelism/5g-business-models/4-when-use.yml",
@@ -146728,6 +146740,9 @@
       "canonical_url": "https://docs.microsoft.com/en-us/learn/modules/5g-business-models/5-knowledge-check",
       "document_id": "57cb13dc-8a68-7370-af0c-01d9f1fa541a",
       "document_version_independent_id": "aecf5c8d-6082-b04e-0a35-c1493f6be244",
+      "github_contributors": [
+        "aysemutlu"
+      ],
       "updated_at": "2021-11-08 12:20 AM",
       "content_git_url": "https://github.com/MicrosoftDocs/learn/blob/main/learn-pr/student-evangelism/5g-business-models/5-knowledge-check.yml",
       "original_content_git_url": "https://github.com/MicrosoftDocs/learn-pr/blob/live/learn-pr/student-evangelism/5g-business-models/5-knowledge-check.yml",
@@ -146795,6 +146810,9 @@
       "canonical_url": "https://docs.microsoft.com/en-us/learn/modules/5g-business-models/6-summary",
       "document_id": "60fd9a53-fc27-ec46-065e-51fb40d7d79e",
       "document_version_independent_id": "6fe4711a-3645-c2e9-741b-d2297d10c2cb",
+      "github_contributors": [
+        "aysemutlu"
+      ],
       "updated_at": "2021-11-08 12:20 AM",
       "content_git_url": "https://github.com/MicrosoftDocs/learn/blob/main/learn-pr/student-evangelism/5g-business-models/6-summary.yml",
       "original_content_git_url": "https://github.com/MicrosoftDocs/learn-pr/blob/live/learn-pr/student-evangelism/5g-business-models/6-summary.yml",
@@ -146849,7 +146867,8 @@
       "document_id": "23e666d0-4576-cb36-55ee-ec5534c7bf94",
       "document_version_independent_id": "53f3014e-5227-772d-c0d1-8a1511d58146",
       "github_contributors": [
-        "ShannonLeavitt"
+        "ShannonLeavitt",
+        "aysemutlu"
       ],
       "updated_at": "2021-11-04 12:22 AM",
       "content_git_url": "https://github.com/MicrosoftDocs/learn/blob/main/learn-pr/student-evangelism/5g-business-models/index.yml",
@@ -352853,6 +352872,9 @@ However, this doesn't mean all Prefab instances have to be identical. You can ov
       "canonical_url": "https://docs.microsoft.com/en-us/learn/modules/create-image-recognition-solution-iot-edge-cognitive-services/5-understand-project-structure",
       "document_id": "133df2e4-f3c3-a458-4fdd-92de95abe6c1",
       "document_version_independent_id": "6cc4b154-4d4a-8fd1-bea4-38d2e98e4000",
+      "github_contributors": [
+        "aysemutlu"
+      ],
       "updated_at": "2021-11-08 12:20 AM",
       "content_git_url": "https://github.com/MicrosoftDocs/learn/blob/main/learn-pr/student-evangelism/create-image-recognition-solution-iot-edge-cognitive-services/5-understand-project-structure.yml",
       "original_content_git_url": "https://github.com/MicrosoftDocs/learn-pr/blob/live/learn-pr/student-evangelism/create-image-recognition-solution-iot-edge-cognitive-services/5-understand-project-structure.yml",
@@ -352911,6 +352933,9 @@ However, this doesn't mean all Prefab instances have to be identical. You can ov
       "canonical_url": "https://docs.microsoft.com/en-us/learn/modules/create-image-recognition-solution-iot-edge-cognitive-services/6-exercise-build-deploy-solution",
       "document_id": "34c2e6f5-1b42-b8a2-53b0-c088e8888a2b",
       "document_version_independent_id": "5e12b198-0650-89ca-1bc5-683f6b29de39",
+      "github_contributors": [
+        "aysemutlu"
+      ],
       "updated_at": "2021-11-08 12:20 AM",
       "content_git_url": "https://github.com/MicrosoftDocs/learn/blob/main/learn-pr/student-evangelism/create-image-recognition-solution-iot-edge-cognitive-services/6-exercise-build-deploy-solution.yml",
       "original_content_git_url": "https://github.com/MicrosoftDocs/learn-pr/blob/live/learn-pr/student-evangelism/create-image-recognition-solution-iot-edge-cognitive-services/6-exercise-build-deploy-solution.yml",
@@ -352969,6 +352994,9 @@ However, this doesn't mean all Prefab instances have to be identical. You can ov
       "canonical_url": "https://docs.microsoft.com/en-us/learn/modules/create-image-recognition-solution-iot-edge-cognitive-services/7-monitor-solution",
       "document_id": "e90d3905-4d8a-3ff1-e7b1-0d5b9527a11b",
       "document_version_independent_id": "8d566f84-44f2-ab33-725c-22a78f5faafb",
+      "github_contributors": [
+        "aysemutlu"
+      ],
       "updated_at": "2021-11-08 12:20 AM",
       "content_git_url": "https://github.com/MicrosoftDocs/learn/blob/main/learn-pr/student-evangelism/create-image-recognition-solution-iot-edge-cognitive-services/7-monitor-solution.yml",
       "original_content_git_url": "https://github.com/MicrosoftDocs/learn-pr/blob/live/learn-pr/student-evangelism/create-image-recognition-solution-iot-edge-cognitive-services/7-monitor-solution.yml",
@@ -353027,6 +353055,9 @@ However, this doesn't mean all Prefab instances have to be identical. You can ov
       "canonical_url": "https://docs.microsoft.com/en-us/learn/modules/create-image-recognition-solution-iot-edge-cognitive-services/8-summary",
       "document_id": "a0b3ed33-e473-f1c5-d3f0-ba29a3952104",
       "document_version_independent_id": "1f6666c4-0c18-c943-3950-51195b47161f",
+      "github_contributors": [
+        "aysemutlu"
+      ],
       "updated_at": "2021-11-08 12:20 AM",
       "content_git_url": "https://github.com/MicrosoftDocs/learn/blob/main/learn-pr/student-evangelism/create-image-recognition-solution-iot-edge-cognitive-services/8-summary.yml",
       "original_content_git_url": "https://github.com/MicrosoftDocs/learn-pr/blob/live/learn-pr/student-evangelism/create-image-recognition-solution-iot-edge-cognitive-services/8-summary.yml",
@@ -353085,6 +353116,9 @@ However, this doesn't mean all Prefab instances have to be identical. You can ov
       "canonical_url": "https://docs.microsoft.com/en-us/learn/modules/create-image-recognition-solution-iot-edge-cognitive-services/9-knowledge-check",
       "document_id": "4d5be5f6-bd64-d685-b2fd-df2d39c7dd49",
       "document_version_independent_id": "43f3380b-6f4e-aa73-2feb-b074099fe052",
+      "github_contributors": [
+        "aysemutlu"
+      ],
       "updated_at": "2021-11-08 12:20 AM",
       "content_git_url": "https://github.com/MicrosoftDocs/learn/blob/main/learn-pr/student-evangelism/create-image-recognition-solution-iot-edge-cognitive-services/9-knowledge-check.yml",
       "original_content_git_url": "https://github.com/MicrosoftDocs/learn-pr/blob/live/learn-pr/student-evangelism/create-image-recognition-solution-iot-edge-cognitive-services/9-knowledge-check.yml",
@@ -405567,7 +405601,8 @@ However, this doesn't mean all Prefab instances have to be identical. You can ov
       "document_version_independent_id": "2a8e73b7-faf9-e0ed-6ec5-43585e765e5f",
       "github_contributors": [
         "wwlpublish",
-        "Krumelur"
+        "Krumelur",
+        "aysemutlu"
       ],
       "updated_at": "2021-12-22 12:22 AM",
       "content_git_url": "https://github.com/MicrosoftDocs/learn/blob/main/learn-pr/student-evangelism/deploy-prebuilt-module-edge-device/1-introduction.yml",
@@ -405630,7 +405665,8 @@ However, this doesn't mean all Prefab instances have to be identical. You can ov
       "github_contributors": [
         "wwlpublish",
         "Krumelur",
-        "megvanhuygen"
+        "megvanhuygen",
+        "aysemutlu"
       ],
       "updated_at": "2021-12-22 12:22 AM",
       "content_git_url": "https://github.com/MicrosoftDocs/learn/blob/main/learn-pr/student-evangelism/deploy-prebuilt-module-edge-device/2-what-is-prebuilt-module.yml",
@@ -405692,7 +405728,8 @@ However, this doesn't mean all Prefab instances have to be identical. You can ov
       "document_version_independent_id": "f97fec45-3d28-cc2b-7b4b-788fd2b8d91f",
       "github_contributors": [
         "wwlpublish",
-        "Krumelur"
+        "Krumelur",
+        "aysemutlu"
       ],
       "updated_at": "2021-12-22 12:22 AM",
       "content_git_url": "https://github.com/MicrosoftDocs/learn/blob/main/learn-pr/student-evangelism/deploy-prebuilt-module-edge-device/3-how-setup-communication.yml",
@@ -405754,7 +405791,8 @@ However, this doesn't mean all Prefab instances have to be identical. You can ov
       "document_version_independent_id": "4ae14d2b-295b-3bcd-7776-b380e0419808",
       "github_contributors": [
         "wwlpublish",
-        "Krumelur"
+        "Krumelur",
+        "aysemutlu"
       ],
       "updated_at": "2022-02-22 12:21 AM",
       "content_git_url": "https://github.com/MicrosoftDocs/learn/blob/main/learn-pr/student-evangelism/deploy-prebuilt-module-edge-device/4-exercise-setup-communication.yml",
@@ -405816,7 +405854,8 @@ However, this doesn't mean all Prefab instances have to be identical. You can ov
       "document_version_independent_id": "13c27963-157d-5fbd-07cf-03e29863b72e",
       "github_contributors": [
         "wwlpublish",
-        "Krumelur"
+        "Krumelur",
+        "aysemutlu"
       ],
       "updated_at": "2021-12-22 12:22 AM",
       "content_git_url": "https://github.com/MicrosoftDocs/learn/blob/main/learn-pr/student-evangelism/deploy-prebuilt-module-edge-device/5-how-deploy-prebuilt-module.yml",
@@ -405878,7 +405917,8 @@ However, this doesn't mean all Prefab instances have to be identical. You can ov
       "document_version_independent_id": "8594dd85-2a7d-4bd0-f68f-2e9e1e98b218",
       "github_contributors": [
         "wwlpublish",
-        "Krumelur"
+        "Krumelur",
+        "aysemutlu"
       ],
       "updated_at": "2021-12-22 12:22 AM",
       "content_git_url": "https://github.com/MicrosoftDocs/learn/blob/main/learn-pr/student-evangelism/deploy-prebuilt-module-edge-device/6-exercise-deploying-prebuilt-module.yml",
@@ -405940,7 +405980,8 @@ However, this doesn't mean all Prefab instances have to be identical. You can ov
       "document_version_independent_id": "1bc8d485-4ced-ce6d-4f16-cb9307ac61c5",
       "github_contributors": [
         "wwlpublish",
-        "Krumelur"
+        "Krumelur",
+        "aysemutlu"
       ],
       "updated_at": "2021-12-22 12:22 AM",
       "content_git_url": "https://github.com/MicrosoftDocs/learn/blob/main/learn-pr/student-evangelism/deploy-prebuilt-module-edge-device/7-knowledge-check.yml",
@@ -406011,7 +406052,8 @@ However, this doesn't mean all Prefab instances have to be identical. You can ov
       "document_version_independent_id": "d75f0686-8499-f10d-1811-c6427c69aac2",
       "github_contributors": [
         "wwlpublish",
-        "Krumelur"
+        "Krumelur",


```

</details>

**Comment by herohua** [comment]
<details><summary>⚠<a href='https://github.com/MicrosoftDocs/VBA-Docs'>VBA-Docs</a>(00:01:03.5295475, 6686 more diff, 2190MB)</summary>

```diff
diff --git "a/D:/docfx.TestData\\regression-test.VBA-Docs\\output/.errors.log" "b/D:/docfx.TestData\\regression-test.VBA-Docs\\.temp/.errors.log"
index 4013960..24e844d 100644
--- "a/D:/docfx.TestData\\regression-test.VBA-Docs\\output/.errors.log"
+++ "b/D:/docfx.TestData\\regression-test.VBA-Docs\\.temp/.errors.log"
@@ -83667,7 +83667,8 @@
   "ms.author": "o365devx",
   "ms.prod": "outlook",
   "ms.technology": "vba",
-  "ms.topic": "conceptual"
+  "ms.topic": "conceptual",
+  "document_url": "https://review.docs.microsoft.com/en-us/help/contribute/validation-ref/image-inline?branch=main"
 }
 {
   "message_severity": "suggestion",
@@ -83682,7 +83683,8 @@
   "ms.author": "o365devx",
   "ms.prod": "outlook",
   "ms.technology": "vba",
-  "ms.topic": "conceptual"
+  "ms.topic": "conceptual",
+  "document_url": "https://review.docs.microsoft.com/en-us/help/contribute/validation-ref/image-inline?branch=main"
 }
 {
   "message_severity": "suggestion",
@@ -83697,7 +83699,8 @@
   "ms.author": "o365devx",
   "ms.prod": "outlook",
   "ms.technology": "vba",
-  "ms.topic": "conceptual"
+  "ms.topic": "conceptual",
+  "document_url": "https://review.docs.microsoft.com/en-us/help/contribute/validation-ref/image-inline?branch=main"
 }
 {
   "message_severity": "suggestion",
@@ -83712,7 +83715,8 @@
   "ms.author": "o365devx",
   "ms.prod": "outlook",
   "ms.technology": "vba",
-  "ms.topic": "conceptual"
+  "ms.topic": "conceptual",
+  "document_url": "https://review.docs.microsoft.com/en-us/help/contribute/validation-ref/image-inline?branch=main"
 }
 {
   "message_severity": "suggestion",
@@ -83727,7 +83731,8 @@
   "ms.author": "o365devx",
   "ms.prod": "outlook",
   "ms.technology": "vba",
-  "ms.topic": "conceptual"
+  "ms.topic": "conceptual",
+  "document_url": "https://review.docs.microsoft.com/en-us/help/contribute/validation-ref/image-inline?branch=main"
 }
 {
   "message_severity": "suggestion",
@@ -83742,7 +83747,8 @@
   "ms.author": "o365devx",
   "ms.prod": "outlook",
   "ms.technology": "vba",
-  "ms.topic": "conceptual"
+  "ms.topic": "conceptual",
+  "document_url": "https://review.docs.microsoft.com/en-us/help/contribute/validation-ref/image-inline?branch=main"
 }
 {
   "message_severity": "suggestion",
@@ -83757,7 +83763,8 @@
   "ms.author": "o365devx",
   "ms.prod": "outlook",
   "ms.technology": "vba",
-  "ms.topic": "conceptual"
+  "ms.topic": "conceptual",
+  "document_url": "https://review.docs.microsoft.com/en-us/help/contribute/validation-ref/image-inline?branch=main"
 }
 {
   "message_severity": "suggestion",
@@ -83772,7 +83779,8 @@
   "ms.author": "o365devx",
   "ms.prod": "outlook",
   "ms.technology": "vba",
-  "ms.topic": "conceptual"
+  "ms.topic": "conceptual",
+  "document_url": "https://review.docs.microsoft.com/en-us/help/contribute/validation-ref/image-inline?branch=main"
 }
 {
   "message_severity": "suggestion",
@@ -83787,7 +83795,8 @@
   "ms.author": "o365devx",
   "ms.prod": "outlook",
   "ms.technology": "vba",
-  "ms.topic": "conceptual"
+  "ms.topic": "conceptual",
+  "document_url": "https://review.docs.microsoft.com/en-us/help/contribute/validation-ref/image-inline?branch=main"
 }
 {
   "message_severity": "suggestion",
@@ -83802,7 +83811,8 @@
   "ms.author": "o365devx",
   "ms.prod": "outlook",
   "ms.technology": "vba",
-  "ms.topic": "conceptual"
+  "ms.topic": "conceptual",
+  "document_url": "https://review.docs.microsoft.com/en-us/help/contribute/validation-ref/image-inline?branch=main"
 }
 {
   "message_severity": "suggestion",
@@ -83817,7 +83827,8 @@
   "ms.author": "o365devx",
   "ms.prod": "project-server",
   "ms.technology": "vba",
-  "ms.topic": "conceptual"
+  "ms.topic": "conceptual",
+  "document_url": "https://review.docs.microsoft.com/en-us/help/contribute/validation-ref/image-inline?branch=main"
 }
 {
   "message_severity": "suggestion",
@@ -83832,7 +83843,8 @@
   "ms.author": "o365devx",
   "ms.prod": "office",
   "ms.technology": "vba",
-  "ms.topic": "conceptual"
+  "ms.topic": "conceptual",
+  "document_url": "https://review.docs.microsoft.com/en-us/help/contribute/validation-ref/image-inline?branch=main"
 }
 {
   "message_severity": "suggestion",
@@ -83847,7 +83859,8 @@
   "ms.author": "o365devx",
   "ms.prod": "office",
   "ms.technology": "vba",
-  "ms.topic": "conceptual"
+  "ms.topic": "conceptual",
+  "document_url": "https://review.docs.microsoft.com/en-us/help/contribute/validation-ref/image-inline?branch=main"
 }
 {
   "message_severity": "suggestion",
@@ -83862,7 +83875,8 @@
   "ms.author": "o365devx",
   "ms.prod": "access",
   "ms.technology": "vba",
-  "ms.topic": "conceptual"
+  "ms.topic": "conceptual",
+  "document_url": "https://review.docs.microsoft.com/en-us/help/contribute/validation-ref/image-inline?branch=main"
 }
 {
   "message_severity": "suggestion",
@@ -83877,7 +83891,8 @@
   "ms.author": "o365devx",
   "ms.prod": "access",
   "ms.technology": "vba",
-  "ms.topic": "conceptual"
+  "ms.topic": "conceptual",
+  "document_url": "https://review.docs.microsoft.com/en-us/help/contribute/validation-ref/image-inline?branch=main"
 }
 {
   "message_severity": "suggestion",
@@ -83892,7 +83907,8 @@
   "ms.author": "o365devx",
   "ms.prod": "access",
   "ms.technology": "vba",
-  "ms.topic": "conceptual"
+  "ms.topic": "conceptual",
+  "document_url": "https://review.docs.microsoft.com/en-us/help/contribute/validation-ref/image-inline?branch=main"
 }
 {
   "message_severity": "suggestion",
@@ -83907,7 +83923,8 @@
   "ms.author": "o365devx",
   "ms.prod": "access",
   "ms.technology": "vba",
-  "ms.topic": "conceptual"
+  "ms.topic": "conceptual",
+  "document_url": "https://review.docs.microsoft.com/en-us/help/contribute/validation-ref/image-inline?branch=main"
 }
 {
   "message_severity": "suggestion",
@@ -83922,7 +83939,8 @@
   "ms.author": "o365devx",
   "ms.prod": "access",
   "ms.technology": "vba",
-  "ms.topic": "conceptual"
+  "ms.topic": "conceptual",
+  "document_url": "https://review.docs.microsoft.com/en-us/help/contribute/validation-ref/image-inline?branch=main"
 }
 {
   "message_severity": "suggestion",
@@ -83937,7 +83955,8 @@
   "ms.author": "o365devx",
   "ms.prod": "visio",
   "ms.technology": "vba",
-  "ms.topic": "reference"
+  "ms.topic": "reference",
+  "document_url": "https://review.docs.microsoft.com/en-us/help/contribute/validation-ref/image-inline?branch=main"
 }
 {
   "message_severity": "suggestion",
@@ -83952,7 +83971,8 @@
   "ms.author": "o365devx",
   "ms.prod": "visio",
   "ms.technology": "vba",
-  "ms.topic": "reference"
+  "ms.topic": "reference",


```

</details>

**Comment by herohua** [comment]
<details><summary>🧭<a href='https://github.com/MicrosoftDocs/azure-docs-pr'>azure-docs-pr</a>(00:02:11.5755622 | exceed 100s, 8025MB)</summary>

```csharp
 28% | docfx!Microsoft.Docs.Build.ParallelUtility+<>c__DisplayClass2_0`1[System.__Canon].<ForEach>b__0(!0)
 23% | docfx!Microsoft.Docs.Build.DocsetBuilder+<>c__DisplayClass41_1.<Build>b__15(class Microsoft.Docs.Build.FilePath)
 21% | docfx!Microsoft.Docs.Build.PageBuilder.Build(class Microsoft.Docs.Build.ErrorBuilder,class Microsoft.Docs.Build.FilePath)
 18% | Microsoft.Docs.MarkdigExtensions!Microsoft.Docs.MarkdigExtensions.ResolveLinkExtension.UpdateLinks(class Markdig.Syntax.MarkdownObject)
 15% | docfx!Microsoft.Docs.Build.PageBuilder.LoadMarkdown(class Microsoft.Docs.Build.ErrorBuilder,class Microsoft.Docs.Build.FilePath)
 13% | docfx!Microsoft.Docs.Build.MarkdownEngine.ToHtml(class Microsoft.Docs.Build.ErrorBuilder,class System.String,class Microsoft.Docs.Build.SourceInfo,value class Microsoft.Docs.Build.MarkdownPipelineType,class Microsoft.Docs.Build.ConceptualModel,bool)
 12% | Markdig!Markdig.Markdown.ToHtml(class System.String,class Markdig.MarkdownPipeline)
  9% | Markdig!Markdig.Parsers.MarkdownParser.Parse()
  8% | docfx!Microsoft.Docs.Build.MarkdigUtility.<Replace>g__ReplaceCore|7_0(class Markdig.Syntax.MarkdownObject,class System.Func`2<class Markdig.Syntax.MarkdownObject,class Markdig.Syntax.MarkdownObject>)
  7% | Markdig!Markdig.Renderers.RendererBase.Write(class Markdig.Syntax.MarkdownObject)
  6% | docfx!Microsoft.Docs.Build.Docfx+<>c__DisplayClass6_0.<CreateCommand>b__0(class Microsoft.Docs.Build.CommandLineOptions)
  6% | docfx!Microsoft.Docs.Build.Docfx+<>c__DisplayClass4_0.<BuildCommand>b__0(class Microsoft.Docs.Build.CommandLineOptions)
  6% | docfx!Microsoft.Docs.Build.Builder.Run(class Microsoft.Docs.Build.CommandLineOptions,class Microsoft.Docs.Build.Package)
  6% | docfx!Microsoft.Docs.Build.Builder.Build(class Microsoft.Docs.Build.ErrorBuilder,class System.IProgress`1<class System.String>,class System.String[])
  4% | docfx!Microsoft.Docs.Build.MarkdigUtility.Visit(class Markdig.Syntax.MarkdownObject,class System.Func`2<class Markdig.Syntax.MarkdownObject,bool>)
  3% | docfx!Microsoft.Docs.Build.Builder+<>c__DisplayClass8_0.<Build>b__0(class Microsoft.Docs.Build.DocsetBuilder)
  3% | docfx!Microsoft.Docs.Build.DocsetBuilder.Build(class System.String[])
  3% | docfx!Microsoft.Docs.Build.PageBuilder.CreatePageOutput(class Microsoft.Docs.Build.ErrorBuilder,class Microsoft.Docs.Build.FilePath,class Newtonsoft.Json.Linq.JObject)
  3% | docfx!Microsoft.Docs.Build.LinkResolver.ResolveLink(value class Microsoft.Docs.Build.SourceInfo`1<class System.String>,class Microsoft.Docs.Build.FilePath,class Microsoft.Docs.Build.FilePath,class Microsoft.Docs.Validation.LinkNode,class System.String)
  3% | docfx!Microsoft.Docs.Build.ParallelUtility.ForEach(class Microsoft.Docs.Build.LogScope,class Microsoft.Docs.Build.ErrorBuilder,class System.Collections.Generic.IEnumerable`1<!!0>,class System.Action`1<!!0>)
  3% | docfx!Microsoft.Docs.Build.MarkdownEngine.GetLink(class Microsoft.Docs.Build.LinkInfo)
  3% | Markdig!Markdig.Renderers.RendererBase.WriteChildren(class Markdig.Syntax.ContainerBlock)
  3% | docfx!Microsoft.Docs.Build.Watch`1[System.__Canon].get_Value()
  2% | docfx!Microsoft.Docs.Build.LinkResolver.TryResolveAbsoluteLink(value class Microsoft.Docs.Build.SourceInfo`1<class System.String>,class Microsoft.Docs.Build.FilePath,class Microsoft.Docs.Build.FilePath,class Microsoft.Docs.Validation.LinkNode,class System.String)
  2% | docfx!Microsoft.Docs.Build.TocLoader+<>c__DisplayClass22_0.<LoadTocNodes>b__0(int32)
  2% | docfx!Microsoft.Docs.Build.MarkdownEngine.GetImageLink(class System.String,class Markdig.Syntax.MarkdownObject,class System.String,class System.String)
  2% | docfx!Microsoft.Docs.Build.TocLoader.LoadTocNode(value class Microsoft.Docs.Build.SourceInfo`1<class Microsoft.Docs.Build.TocNode>,class Microsoft.Docs.Build.FilePath,class Microsoft.Docs.Build.FilePath,class System.Collections.Generic.List`1<class Microsoft.Docs.Build.FilePath>,class System.Collections.Generic.List`1<class Microsoft.Docs.Build.FilePath>)
  2% | docfx!Microsoft.Docs.Build.DocsValidationExtension+<>c__DisplayClass0_0.<UseDocsValidation>b__0(class Markdig.Syntax.MarkdownDocument)
  2% | docfx!Microsoft.Docs.Build.ContentValidator.ValidateLink(class Microsoft.Docs.Build.FilePath,class Microsoft.Docs.Validation.LinkNode)
  2% | docfx!Microsoft.Docs.Build.ErrorBuilder.AddRange(class System.Collections.Generic.IEnumerable`1<class Microsoft.Docs.Build.Error>)

```

</details>

**Pull request merged**